benchalerts/benchalerts/conbench_dataclasses.py

```diff
@@ -190,14 +186,34 @@ class FullComparisonInfo:
     the comparisons to their baselines.
     """
 
+    # Can be empty.
     run_comparisons: List[RunComparisonInfo]
 
-    def __post_init__(self):
-        # Any code that constructs an instance should ensure run_comparisons is non-empty
-        assert self.run_comparisons
+    @property
+    def has_any_contender_runs(self) -> bool:
+        """Whether there are any contender runs available."""
+        return bool(self.run_comparisons)
+
+    @property
+    def has_any_contender_results(self) -> bool:
+        """Whether there are any contender benchmark results available."""
+        for run_comparison in self.run_comparisons:
+            if run_comparison.contender_benchmark_result_info:
+                return True
+        return False
+
+    @property
+    def has_any_z_analyses(self) -> bool:
+        """Whether there are any lookback z-score analyses available."""
+        for run_comparison in self.run_comparisons:
+            if run_comparison.compare_results:
+                for compare_result in run_comparison.compare_results:
+                    if compare_result["analysis"]["lookback_z_score"]:
+                        return True
+        return False
```

Comment:
A useless comment on code style.

I _like_ this style here.

Not sure how to put it into words: saving the `else:` clauses, having the catch-all-other-cases return statement.

Often we/people write code in the inverse shape. Which is not wrong, but this one here is just so much more problem-oriented and elegant I think.

Reply:
I do too! I wrote a few of these almost completely with GitHub Copilot 🙃 🤖

Reply:
>  with GitHub Copilot upside_down_face robot

haha! Thanks for the punch, I still want to be proud of my craft sometimes :cry: Took years to converge to this style (to see more often when it's possible _after_ having written the other variant, or to start with that right away). Anyway :)